lib/npm.js

```diff
@@ -113,12 +113,18 @@ class Npm extends EventEmitter {
     }
 
     const workspacesEnabled = this.config.get('workspaces')
+    /* istanbul ignore next */
+    const implicitWorkspace = (this.config.get('workspace', 'default') || []).length > 0
     const workspacesFilters = this.config.get('workspace')
     if (workspacesEnabled === false && workspacesFilters.length > 0) {
       throw new Error('Can not use --no-workspaces and --workspace at the same time')
     }
 
-    const filterByWorkspaces = workspacesEnabled || workspacesFilters.length > 0
+    // only call execWorkspaces when we have workspaces explicitly set
+    // or when it is implicit and not in our ignore list
+    const filterByWorkspaces =
+      (workspacesEnabled || workspacesFilters.length > 0)
+      && (!implicitWorkspace || !command.ignoreImplicitWorkspace)
```

Comment:
Adjust the comment:
```suggestion
    // only call execWorkspaces when we have workspaces explicitly set
    // or when it is implicit or marked as ignore implicit workspace
    const filterByWorkspaces =
      (workspacesEnabled || workspacesFilters.length > 0)
      && (!implicitWorkspace || !command.ignoreImplicitWorkspace)
```


Or rename `ignoreImplicitWorkspace` to `ignoreWorkspace` which I think it's more representative:
```suggestion
    // only call execWorkspaces when the command doesn't ignore workspaces,
    // workspace is not implicitly set and we have workspaces explicitly set
    const filterByWorkspaces =
      !command.ignoreWorkspace && !implicitWorkspace && (workspacesEnabled || workspacesFilters.length > 0)
```

Reply:
And `implicitWorkspace` shouldn't conflict with explicit workspace? Or explicit shouldn't take precedence?

Reply:
There's a lot of ways we could slice it. Right now, if your cwd is a workspace, it'll run the command and ignore any explicit workspace you specify. This makes sense for something like `bin`, which has always worked on cwd, but it could be confusing for other commands. If you're not in a workspace and and you specify one, many commands will still fail.

We could go back and forth on the best way to handle these edge cases, but ultimately, I'd rather ship and iterate. If there are cases and commands that need more tweaking, we could patch the behavior again. But for now, this gets us back to running commands that don't really care about workspaces, but generally still errors if you try to explicitly set a workspace on a command that doesn't support them.